### ISSUE TITLE
NPCs with responsive emote behaviour

### ISSUE THREAD

**Issue by LordUsagi** [OregonCore/OregonCore#1251]
The below list are creatures who are known to respond to certain emotes but currently do not. This generally includes all barmaids, but I haven't listed them all here.
- [x] Kylene, Worlds End Tavern, Shattrath
- [x] Johnathan Marcus, Stormwind

**Comment by Zaffy** [comment]
Do we also have the responses?

**Comment by Lumber-Jack-2** [comment]
the response is a simple emote by the creature itself

**Comment by dikkedeur** [comment]
like when you wave to guards, they wave back? or was it only jonathan marcus?

found this on tc code...

`
void DoReplyToTextEmote(uint32 emote)
        {
            switch (emote)
            {
                case TEXT_EMOTE_KISS:
                    me->HandleEmoteCommand(EMOTE_ONESHOT_BOW);
                    break;

```
            case TEXT_EMOTE_WAVE:
                me->HandleEmoteCommand(EMOTE_ONESHOT_WAVE);
                break;

            case TEXT_EMOTE_SALUTE:
                me->HandleEmoteCommand(EMOTE_ONESHOT_SALUTE);
                break;

            case TEXT_EMOTE_SHY:
                me->HandleEmoteCommand(EMOTE_ONESHOT_FLEX);
                break;

            case TEXT_EMOTE_RUDE:
            case TEXT_EMOTE_CHICKEN:
                me->HandleEmoteCommand(EMOTE_ONESHOT_POINT);
                break;
        }
    }

    void ReceiveEmote(Player* player, uint32 textEmote) override
    {
        switch (me->GetEntry())
        {
            case NPC_STORMWIND_CITY_GUARD:
            case NPC_STORMWIND_CITY_PATROLLER:
            case NPC_ORGRIMMAR_GRUNT:
                break;
            default:
                return;
        }

        if (!me->IsFriendlyTo(player))
            return;

        DoReplyToTextEmote(textEmote);
    }
```

`

**Comment by dikkedeur** [comment]
the moment you find out that above is already in oc and working COUGH!!!! :)

**Comment by LordUsagi** [comment]
Not just waving, for example Cenarion Refuge guards passively salute players exalted walking by them. We recently did the silvermoon guardians flirting/flexing/kissing/ruding back to players. It's a lot more then just a wave in some cases, but in other cases its just a wave. I'm trying to compile a list. Otherwise wowhead will have the necessary details. Apparently ALL barmaids had similar behaviour to Kylene.

**Comment by delebota** [comment]
I got info for Kylene here: http://wowwiki.wikia.com/wiki/Kylene
The only other one I could only find info for was Elly Langston, which I found here: http://www.wowhead.com/npc=1328/elly-langston#comments:id=2074495
I applied the options for Elly to Gwenna Firebrew and Edris Barleybeard as well. 

I also noticed that Kylene seems to spawn in a dancing state which isn't correct. Also I'm pretty sure she walked around inside the tavern on retail which she does not currently do either. Those should probably go under a separate issue though.
